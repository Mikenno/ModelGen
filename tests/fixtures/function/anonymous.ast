 Module
- Procedure
-- Invalid
-- Tuple
-- Call
--- Identifier print
--- String Anonymous

- Call
-- Procedure
--- Invalid
--- Tuple
--- Call
---- Identifier print
---- String Anonymous

- Call
-- Procedure
--- Invalid
--- Tuple
--- Call
---- Identifier print
---- String Anonymous

- Call
-- Procedure
--- Invalid
--- Tuple
---- Identifier x
---- Identifier y
--- Call
---- Identifier print
---- String Anonymous
---- Identifier x
---- Identifier y
-- Integer 1
-- Integer 2

- Assign
-- Identifier test1
-- Procedure
--- Invalid
--- Tuple
--- Call
---- Identifier print
---- String Anonymous

- Assign
-- Identifier test2
-- Procedure
--- Invalid
--- Tuple
--- Call
---- Identifier print
---- String Anonymous

- Assign
-- Identifier test3
-- Procedure
--- Invalid
--- Tuple
--- Block
---- Call
----- Identifier print
----- String Anonymous
---- Call
----- Identifier print
----- String Anonymous

- Assign
-- Identifier test4
-- Procedure
--- Invalid
--- Tuple
--- Block
---- Call
----- Identifier print
----- String Anonymous
---- Call
----- Identifier print
----- String Anonymous

- Tuple
-- Procedure
--- Invalid
--- Tuple
-- Procedure
--- Invalid
--- Tuple
-- Procedure
--- Invalid
--- Tuple

- Tuple
-- Procedure
--- Invalid
--- Tuple
--- Call
---- Identifier print
---- Integer 1
-- Procedure
--- Invalid
--- Tuple
--- Call
---- Identifier print
---- Integer 2
-- Procedure
--- Invalid
--- Tuple
--- Call
---- Identifier print
---- Integer 3

- Tuple
-- Procedure
--- Identifier test1
--- Tuple
--- Call
---- Identifier print
---- Integer 1
-- Procedure
--- Identifier test2
--- Tuple
--- Call
---- Identifier print
---- Integer 2
-- Procedure
--- Identifier test3
--- Tuple
--- Call
---- Identifier print
---- Integer 3

- Assign
-- Identifier x
-- Tuple
--- Procedure
---- Invalid
---- Tuple
---- Call
----- Identifier print
----- String Anonymous 1
--- Procedure
---- Invalid
---- Tuple
---- Call
----- Identifier print
----- String Anonymous 2
--- Procedure
---- Invalid
---- Tuple
---- Call
----- Identifier print
----- String Anonymous 3

- Assign
-- Identifier y
-- Tuple
--- Procedure
---- Invalid
---- Tuple
---- Call
----- Identifier print
----- String Anonymous 1
--- Procedure
---- Invalid
---- Tuple
---- Call
----- Identifier print
----- String Anonymous 2
--- Procedure
---- Invalid
---- Tuple
---- Call
----- Identifier print
----- String Anonymous 3

- Assign
-- Identifier z
-- Tuple
--- Call
---- Procedure
----- Invalid
----- Tuple
----- Call
------ Identifier print
------ String Anonymous 1
--- Call
---- Procedure
----- Invalid
----- Tuple
----- Call
------ Identifier print
------ String Anonymous 2
--- Call
---- Procedure
----- Invalid
----- Tuple
----- Call
------ Identifier print
------ String Anonymous 3