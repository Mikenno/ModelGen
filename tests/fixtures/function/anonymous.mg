
(proc() print("Anonymous"))

(proc() print("Anonymous"))()
(((proc() print("Anonymous"))))()

(proc(x, y) print("Anonymous", x, y))(1, 2)

test1 = proc() print("Anonymous")
test2 = (proc() print("Anonymous"))
test3 = proc()
	print("Anonymous")
	print("Anonymous")

test4 = (proc()
	print("Anonymous")
	print("Anonymous")
)

(proc(), proc(), proc())
((proc() print(1)), (proc() print(2)), (proc() print(3)))

((proc test1() print(1)), (proc test2() print(2)), (proc test3() print(3)))

x = (
	(proc() print("Anonymous 1")),
	(proc() print("Anonymous 2")),
	(proc() print("Anonymous 3"))
)

y = (
	(proc() print("Anonymous 1")),
	(proc() print("Anonymous 2")),
	(proc() print("Anonymous 3"))
)

z = (
	(proc() print("Anonymous 1"))(),
	(proc() print("Anonymous 2"))(),
	(proc() print("Anonymous 3"))()
)
