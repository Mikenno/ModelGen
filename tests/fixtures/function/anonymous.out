Anonymous
Anonymous
Anonymous 1 2
Anonymous 1
Anonymous 2
Anonymous 3
